R/hiper_glm.R

```diff
@@ -1,23 +1,37 @@
 #' @export
 hiper_glm <- function(design, outcome, model, option = list()){
-  supported_model <- c("linear")
+  supported_model <- c("linear", "logit")
   if (!(model %in% supported_model)) {
     stop(sprintf("The model %s is not supported.", model))
   }
   hglm_out <- list()
   class(hglm_out) <- "hglm"
-  if(model == "linear"){
-    if(is.null(option$mle_solver)){
+  if (model == "linear") {
+    if (is.null(option$mle_solver)) {
       option$mle_solver = "PINV"
     }
-    if(option$mle_solver == "PINV"){
+    if (option$mle_solver == "PINV") {
       hglm_out$coef = lm_pseudo_inv(design, outcome)
     }
-    else if(option$mle_solver == "BFGS"){
+    else if (option$mle_solver == "BFGS") {
       hglm_out$coef = lm_bfgs(design, outcome)
     }
-    else{
-      stop(sprintf("This mle algorithm in lnear model is not supported."))
+    else {
+      stop(print("This mle algorithm in lnear model is not supported."))
+    }
+  }
+  else if (model == "logit") {
+    if (is.null(option$mle_solver)) {
+      option$mle_solver = "Newton"
+    }
+    if (option$mle_solver == "Newton") {
```

Comment:
```suggestion
    if (is.null(option$mle_solver) | option$mle_solver == "Newton") {
```

-1

Something like this will be more concise